library/runCmds.py

```diff
@@ -9,74 +19,81 @@
+# of semaphores involved.
+
+try:
+    from thread import get_ident as _get_ident
+except ImportError:
+    from dummy_thread import get_ident as _get_ident
+from ansible.module_utils.basic import *
 
 t = gettext.translation('rho', 'locale', fallback=True)
 _ = t.ugettext
-#
-# sys.path.insert(0, '/home/kbattula/devel/rho2/')
 
-#!/usr/bin/python
-# Copyright (c) 2009 Red Hat, Inc.
-#
-# This software is licensed to you under the GNU General Public License,
-# version 2 (GPLv2). There is NO WARRANTY for this software, express or
-# implied, including the implied warranties of MERCHANTABILITY or FITNESS
-# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
-# along with this software; if not, see
-# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
-#
 """cmds to run on machines being inventory"""
 
+# RhoCmd and its sub-classes are wrapper classes around
+# the cli cmds we run on the host machines. The
+# rho_cmd class will have five dictionaries. The
+# cmd_names dictionary maps all the commands that are associated
+# with the class to the corresponding fields(facts) requested.
+# The cmd_strings dictionary maps each command name to the
+# actual command string that is run in the cli. The results
+# so obtained are stored in the cmd_results dictionary whose
+# keys are the command names. The fields dictionary stores
+# the localized versions of the field names and the data
+# dictionary is responsible for storing all the facts collected
+# from all the classes ever invoked.
 
 
-
-
-
-# basic idea, wrapper classes around the cli cmds we run on the machines
-# to be inventories. the rho_cmd class will have a string for the
-# actually command to run, a name, a 'data' attrib storing the
-# data retrivied (probably just a dict). I think it makes the most
-# sense to go ahead and parse the results of the remote cmd when we
-# run it and store it then in the data (as opposed to doing it at
-# data read time). I think it's pretty small data sets either way, so
-# not a big deal.
-
-# If we use a dict like class for the data, we maybe able to make
-# the report generation as simple as python string formatting tricks.
-# I'd like to try to avoid type'ing the data fields and just treating
-# everything as strings, since the primary target seems to be csv
-# output.
-
 class RhoCmd(object):
-    name = "base"
-    fields = {}
 
     def __init__(self):
-        #        self.cmd_strings = cmd
-        self.cmd_results = []
+        # self.cmd_strings = cmd
+        self.cmd_results = {}
         self.data = {}
+        self.cmd_strings = {}
+        self.cmd_names = {}
+        self.name = "cmd_name"
+        self.fields = {}
 
-    # we're not actually running the class on the hosts, so
-    # we will need to populate it with the output of the ssh stuff
-    # we can send a list of commands, so we expect output to be a list
-    # of output strings
+    def run_cmd(self, facts):
```

Comment:
The run_cmd method is not performing any validation of the input (cmd_strings, cmd_names, fields, etc)